backend/icpy/agent/helpers.py

```
@@ -69,9 +74,141 @@ def get_openai_token_param(model_name: str, max_tokens: int) -> dict:
+        if content is None:
+            return ""
+        # Primitive -> str
+        return str(content)
+    except Exception:
+        return str(content) if content is not None else ""
+
+
+def normalize_history(
+    raw_history: Any,
+    *,
+    drop_empty_user: bool = True,
```

Ruff (0.13.1)

Unused function argument: `drop_empty_user`

(ARG001)